gui/src/Parsing/ServerParser.cpp

```diff
@@ -0,0 +1,523 @@
+/*
+** EPITECH PROJECT, 2024
+** Zappy GUI
+** File description:
+** Parse server command
+*/
+
+#include "Parsing/ServerParser.hpp"
+
+#include <sstream>
+
+Gui::ServerParser::ServerParser()
+{
+    _functionsMap["msz"] = &_parseCommandMSZ;
+    _functionsMap["bct"] = &_parseCommandBCT;
+    _functionsMap["tna"] = &_parseCommandTNA;
+    _functionsMap["pnw"] = &_parseCommandPNW;
+    _functionsMap["ppo"] = &_parseCommandPPO;
+    _functionsMap["plv"] = &_parseCommandPLV;
+    _functionsMap["pin"] = &_parseCommandPIN;
+    _functionsMap["pex"] = &_parseCommandPEX;
+    _functionsMap["pfk"] = &_parseCommandPFK;
+    _functionsMap["pdr"] = &_parseCommandPDR;
+    _functionsMap["pgt"] = &_parseCommandPGT;
+    _functionsMap["pdi"] = &_parseCommandPDI;
+    _functionsMap["enw"] = &_parseCommandENW;
+    _functionsMap["ebo"] = &_parseCommandEBO;
+    _functionsMap["edi"] = &_parseCommandEDI;
+    _functionsMap["sgt"] = &_parseCommandSGT;
+    _functionsMap["sst"] = &_parseCommandSST;
+    _functionsMap["seg"] = &_parseCommandSEG;
+    _functionsMap["suc"] = &_parseCommandSUC;
+    _functionsMap["sbp"] = &_parseCommandSBP;
+    // TODO : pbc, pic, pie, smg
+    // #15 (https://github.com/FppEpitech/Zappy/issues/15))
+}
+
+std::vector<std::string> Gui::ServerParser::parse(const std::string& command)
+{
+    std::istringstream stream(command);
+    std::string keyCommand;
+    stream >> keyCommand;
+
+    if (_functionsMap[keyCommand] == nullptr)
+        throw Errors::ServerParser("Server's command didn't exist.");
+    return (_functionsMap[keyCommand])(command);
+}
+
+std::vector<std::string> Gui::ServerParser::_parseCommandMSZ(const std::string& command)
+{
+    std::vector<std::string> arguments;
+    std::istringstream stream(command);
+    int x,y;
+    std::string none;
+
+    stream >> none;
+    stream >> x;
+    stream >> y;
+
+    if (stream.fail())
+        throw Errors::ServerParser("Wrong parameters for 'msz' command.");
+    stream >> none;
+    if (!stream.fail())
+        throw Errors::ServerParser("Too many parameters for 'msz' command.");
```

Comment:
```suggestion
    if (stream.peek() != std::istringstream::traits_type::eof()) {
        throw Errors::ServerParser("Too many parameters for 'msz' command.");
    }
```

Use stream.peek() for a clean and clear code.